internal/services/integrationtesting/consistency_test.go

```diff
@@ -427,152 +434,176 @@ func validateLookupSubjects(t *testing.T, vctx validationContext) {
 				subjectType := subjectType
 				t.Run(fmt.Sprintf("%s#%s", subjectType.Namespace, subjectType.Relation),
 					func(t *testing.T) {
-						resolvedSubjects, err := vctx.serviceTester.LookupSubjects(context.Background(), resource, subjectType, vctx.revision, nil)
-						require.NoError(t, err)
-
-						// Ensure the subjects found include those defined as expected. Since the
-						// accessibility set does not include "inferred" subjects (e.g. those with
-						// permissions as their subject relation, or wildcards), this should be a
-						// subset.
-						expectedDefinedSubjects := vctx.accessibilitySet.DirectlyAccessibleDefinedSubjectsOfType(resource, subjectType)
-						requireSubsetOf(t, maps.Keys(resolvedSubjects), maps.Keys(expectedDefinedSubjects))
-
-						// Ensure all subjects in true and caveated assertions for the subject type are found
-						// in the LookupSubject result, except those added via wildcard.
-						for _, parsedFile := range vctx.clusterAndData.Populated.ParsedFiles {
-							for _, entry := range []struct {
-								assertions         []blocks.Assertion
-								requiresPermission bool
-							}{
-								{
-									assertions:         parsedFile.Assertions.AssertTrue,
-									requiresPermission: true,
-								},
-								{
-									assertions:         parsedFile.Assertions.AssertCaveated,
-									requiresPermission: false,
-								},
-							} {
-								for _, assertion := range entry.assertions {
-									assertionRel := tuple.MustFromRelationship[*v1.ObjectReference, *v1.SubjectReference, *v1.ContextualizedCaveat](assertion.Relationship)
-									if !assertionRel.ResourceAndRelation.EqualVT(resource) {
-										continue
+						for _, pageSize := range []uint32{0, 2} {
+							pageSize := pageSize
+							t.Run(fmt.Sprintf("pagesize-%d", pageSize), func(t *testing.T) {
+								// Loop until all subjects have been found or we've hit max iterations.
+								var currentCursor *v1.Cursor
+								resolvedSubjects := map[string]*v1.LookupSubjectsResponse{}
+								for i := 0; i < 100; i++ {
```

Comment:
should this be a for loop without the condition? as it stands it's confusing as we have to iterate indefinitely until all elements are streamed

Reply:
I don't want it to hang indefinitely on failure so I put a "large" limit on it